backend/services/tag_service.py

```
@@ -151,6 +151,60 @@ def build_tag_groups_payload(
+    params = {"existing_id": keep_id, "new_id": dropped_id}
+    if exclude_suggestion_id is not None:
+        params["exclude_id"] = str(exclude_suggestion_id)
+
+    # Any other pending suggestion pointing at dropped_tag as ITS existing_tag would
+    # otherwise be cascade-deleted when dropped_tag is removed — repoint it at
+    # keep_tag so it survives to be approved/rejected later.
+    db.execute(text(f"""
+        UPDATE tag_normalization_suggestions
+        SET existing_tag_id = :existing_id
+        WHERE existing_tag_id = :new_id AND status = 'pending' {exclude_clause}
+    """), params)
```

Ruff (0.16.3)

[error] Possible SQL injection vector through string-based query construction

(S608)